src/lib/components/Editor.svelte

```diff
@@ -132,6 +168,29 @@
     underlineSelection(editor, selections || []);
   }
 
+  const persistentTableStore = getContext(
+    "rill:app:persistent-table-store"
+  ) as PersistentTableStore;
+  const derivedTableStore = getContext(
+    "rill:app:derived-table-store"
+  ) as DerivedTableStore;
+
+  const schema = $persistentTableStore.entities.reduce(
+    (acc, persistentTable: PersistentTableEntity) => {
+      const derivedTable: DerivedTableEntity = $derivedTableStore.entities.find(
+        (derivedTable) => persistentTable.id === derivedTable.id
+      );
+      const columnNames = derivedTable?.profile.map((col) => col.name);
+      return (acc[persistentTable.tableName] = columnNames), acc;
```

Comment:
Any reason to pack both assignment and return here? The actual return value is just acc right? Would be good for readability to split them up.

Reply:
Good catch, just fixed